pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -583,3 +589,53 @@ func (r *ReconcileClusterProvision) deleteInstallJob(provision *hivev1.ClusterPr
 	}
 	return reconcile.Result{}, nil
 }
+
+// getWorkers fetches the number of worker replicas from the installConfig, and reports "unknown" if not available
+func (r *ReconcileClusterProvision) getWorkers(cd hivev1.ClusterDeployment) string {
+	icSecret := &corev1.Secret{}
+	err := r.Get(context.Background(),
+		types.NamespacedName{
+			Namespace: cd.Namespace,
+			Name:      cd.Spec.Provisioning.InstallConfigSecretRef.Name,
+		},
+		icSecret)
+	if err != nil {
+		r.logger.WithError(err).Error("Error loading install config secret")
+		return "unknown"
+	}
+	ic := &installertypes.InstallConfig{}
+	if err := yaml.Unmarshal(icSecret.Data["install-config.yaml"], &ic); err != nil {
+		r.logger.WithError(err).Error("could not unmarshal InstallConfig")
+		return "unknown"
+	}
+	return strconv.FormatInt(*ic.WorkerMachinePool().Replicas, 10)
+}
+
+func (r *ReconcileClusterProvision) logProvisionSuccessFailureMetric(
+	stage hivev1.ClusterProvisionStage, instance *hivev1.ClusterProvision) {
+	cd := &hivev1.ClusterDeployment{}
+	if err := r.Client.Get(context.Background(), types.NamespacedName{Namespace: instance.Namespace,
+		Name: instance.Spec.ClusterDeploymentRef.Name}, cd); err != nil {
+		r.logger.WithError(err).Error("error getting cluster deployment")
+	} else {
+		if stage == hivev1.ClusterProvisionStageFailed {
+			metricInstallFailureSeconds.WithLabelValues(
+				hivemetrics.GetClusterDeploymentType(cd),
+				cd.Labels[hivev1.HiveClusterPlatformLabel],
+				cd.Labels[hivev1.HiveClusterRegionLabel],
+				*cd.Status.InstallVersion,
+				r.getWorkers(*cd),
+				strconv.Itoa(instance.Spec.Attempt),
+			).Observe(time.Since(instance.CreationTimestamp.Time).Seconds())
+		} else {
+			metricInstallSuccessSeconds.WithLabelValues(
+				hivemetrics.GetClusterDeploymentType(cd),
+				cd.Labels[hivev1.HiveClusterPlatformLabel],
+				cd.Labels[hivev1.HiveClusterRegionLabel],
+				*cd.Status.InstallVersion,
+				r.getWorkers(*cd),
+				strconv.Itoa(instance.Spec.Attempt),
+			).Observe(time.Since(instance.CreationTimestamp.Time).Seconds())
+		}
```

Comment:
nit: Could dedup this as:
```suggestion
		theMetric := metricInstallSuccessSeconds
		if stage == hivev1.ClusterProvisionStageFailed {
			theMetric = metricInstallFailureSeconds
		}
		theMetric.WithLabelValues(
			hivemetrics.GetClusterDeploymentType(cd),
			cd.Labels[hivev1.HiveClusterPlatformLabel],
			cd.Labels[hivev1.HiveClusterRegionLabel],
			*cd.Status.InstallVersion,
			r.getWorkers(*cd),
			strconv.Itoa(instance.Spec.Attempt),
		).Observe(time.Since(instance.CreationTimestamp.Time).Seconds())
```